server/src/main/java/org/opensearch/search/ResourceType.java

```diff
@@ -10,21 +10,26 @@
 
 import org.opensearch.common.annotation.PublicApi;
 import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.tasks.resourcetracker.ResourceStats;
+import org.opensearch.tasks.Task;
 
 import java.io.IOException;
+import java.util.function.Function;
 
 /**
  * Enum to hold the resource type
  */
 @PublicApi(since = "2.x")
```

Comment:
out of scope of your PR but this should ideally be a concrete version

Reply:
True, i see a concrete version being used everywhere, can we avoid changes to this in this PR ?

Reply:
Yeah we can take it as a follow-up in a separate PR